### PR TITLE
Improves helm upgrade command

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/1253

### CODE REVIEW

File: service/deployer/installer/helm/helm.go [hunk 1]
```diff
@@ -219,14 +219,15 @@ func (i *HelmInstaller) Install(event eventerspec.DeploymentEvent) error {
 		return microerror.MaskAny(err)
 	}
 
-	if err := i.runHelmCommand(
-		"install",
-		"install",
-		tarballPath,
+	installCommand := []string{
+		"upgrade",
```
Comment (JosephSalisbury) [inline service/deployer/installer/helm/helm.go]: `upgrade --install` essentially creates the release if it's not already installed, or upgrades it if it is. Nice.

File: service/deployer/installer/helm/helm.go [hunk 2]
```diff
@@ -219,14 +219,15 @@ func (i *HelmInstaller) Install(event eventerspec.DeploymentEvent) error {
...
+		"--install",
 		"--values",
 		valuesFile,
-		"--wait",
-	); err != nil {
+		project,
```
Comment (JosephSalisbury) [inline service/deployer/installer/helm/helm.go]: Now naming releases specifically, as opposed to helm's random naming.

File: service/deployer/installer/helm/helm.go (hunk 1 above)
Comment (JosephSalisbury) [inline service/deployer/installer/helm/helm.go]: Beforehand, upgrading was not working - we essentially were trying to install something that was already installed.

File: service/deployer/installer/helm/helm.go (hunk 2 above)
Comment (xh3b4sd) [inline service/deployer/installer/helm/helm.go]: For my interest, what is the random naming and how is it named now?

Comment (JosephSalisbury) [inline service/deployer/installer/helm/helm.go]: Was similar to docker images, like `ferocious-mongoose`. Fun, but needed to go so we could upgrade reliably.

**Pull request merged**